net/portmapper/upnp.go

```diff
@@ -559,6 +559,20 @@ func (c *Client) tryUPnPPortmapWithDevice(
 		return netip.AddrPort{}, nil, err
 	}
 
+	// If we have no client, we cannot continue; this can happen if we get
+	// a valid UPnP response that does not contain any of the service types
+	// that we know how to use.
+	if client == nil {
```

Comment:
can we update docs on `selectBestService()` that it may return `nil, nil` in case this gets called elsewhere in the future?

Reply:
Good call; done 👍